src/promptflow/promptflow/_sdk/_utils.py

```diff
@@ -1126,6 +1127,43 @@ def pd_read_json(file) -> "DataFrame":
         return pd.read_json(f, orient="records", lines=True)
 
 
+def get_mac_address() -> Union[str, None]:
+    """Get the MAC ID of the first network card."""
+    try:
+        import psutil
+
+        mac_address = None
+        net_address = psutil.net_if_addrs()
+        eth = []
+        if "Ethernet" in net_address:  # windows
+            eth = net_address["Ethernet"]
+        elif "eth0" in net_address:  # linux & mac
+            eth = net_address["eth0"]
+        for net_interface in eth:
+            if net_interface.family == psutil.AF_LINK:  # mac address
+                mac_address = str(net_interface.address)
+                break
+
+        # If obtaining the network card MAC ID fails, obtain other MAC ID
+        if mac_address is None:
+            node = uuid.getnode()
+            if node != 0:
+                mac_address = str(uuid.UUID(int=node).hex[-12:])
+
+        return mac_address
+    except Exception as e:
+        logger.debug(f"get mac id error: {str(e)}")
+        return None
+
+
+def gen_uuid_by_mac_id() -> Union[str, None]:
+    mac_address = get_mac_address()
+    if mac_address:
+        mac_address_hash = hashlib.sha256(mac_address.encode()).hexdigest()
```

Comment:
Not sure if we should confirm with privacy team about this

Reply:
ok, I will go and confirm.